src/AbstractHyperrectangle.jl

```diff
@@ -293,3 +295,48 @@ The lower coordinate of the hyperrectangular set in the given dimension.
 function low(H::AbstractHyperrectangle{N}, i::Int)::N where {N<:Real}
     return center(H)[i] - radius_hyperrectangle(H, i)
 end
+
+"""
+    partition(H::AbstractHyperrectangle{N}, i_blocks::AbstractVector{Int}
+             ) where {N<:Real}
+
+Partition a hyperrectangular set into uniform sub-hyperrectangles.
+
+### Input
+
+- `H`        -- hyperrectangular set
+- `i_blocks` -- number of blocks in the partition for each dimension
+
+### Output
+
+A list of `Hyperrectangle`s.
+"""
+function partition(H::AbstractHyperrectangle{N}, i_blocks::AbstractVector{Int}
+                  )::Vector{Hyperrectangle{N}} where {N<:Real}
```

Comment:
shall we assert that `length(i_blocks) == dim(H)`?

also i rename `i_blocks` to `num_blocks`?